### ISSUE TITLE
503 response in /trades for a single IP only

### ISSUE THREAD

**Issue by bartekn** [stellar/go#3173]
### What version are you using?

1.10.1

### What did you do?

There's a weird behaviour affecting a single IP streaming `/trades` endpoint. It sends streaming requests every 10 seconds, `/trades` responds with `503` HTTP status after 10 sec. I tried to reproduce the behaviour with `ab` (also sending streaming requests) and all responses were successful (`200` HTTP status):
```
Percentage of the requests served within a certain time (ms)
  50%    668
  66%    678
  75%    696
  80%    698
  90%    701
  95%    704
  98%    709
  99%    730
 100%    730 (longest request)
```

The underlying error is: `canceling statement due to user request`.

### What did you expect to see?

The error indicates the cancelled request that can be triggered by timeout middleware. However, I'm not sure why the request cannot be finished it it works for me. If this is a client issue (like reading response too slowly) it should be a 4xx error.

### What did you see instead?

`200 OK` response.

**Comment by bartekn** [comment]
This can be an instance of https://github.com/stellar/go/issues/3710.

**Comment by bartekn** [comment]
Fixed in #3710.
